Trading/Mode/staggered_orders_trading_mode/staggered_orders_trading_mode.py

```diff
@@ -0,0 +1,845 @@
+                                                                                          symbol_market):
+                selling = order_data.side == TradeOrderSide.SELL
+                if selling:
+                    if get_portfolio_currency(self.exchange_manager, currency) < order_quantity:
+                        return None
+                elif get_portfolio_currency(self.exchange_manager, market) < order_quantity * order_price:
+                    return None
+                order_type = TraderOrderType.SELL_LIMIT if selling else TraderOrderType.BUY_LIMIT
+                current_order = create_order_instance(trader=self.exchange_manager.trader,
+                                                      order_type=order_type,
+                                                      symbol=order_data.symbol,
+                                                      current_price=current_price,
+                                                      quantity=order_quantity,
+                                                      price=order_price)
+                created_order = await self.exchange_manager.trader.create_order(current_order)
+        except InsufficientFunds as e:
+            raise e
+        except Exception as e:
+            self.logger.error(f"Failed to create order : {e}. Order: {order_data}")
+            self.logger.exception(e)
+            return None
+        return created_order
+
+
+class StaggeredOrdersTradingModeProducer(AbstractTradingModeProducer):
+
+    FILL = 1
+    ERROR = 2
+    NEW = 3
+    min_quantity = "min_quantity"
+    max_quantity = "max_quantity"
+    min_cost = "min_cost"
+    max_cost = "max_cost"
+    min_price = "min_price"
+    max_price = "max_price"
+    PRICE_FETCHING_TIMEOUT = 60
+    # health check once every 3 days
+    HEALTH_CHECK_INTERVAL_SECS = DAYS_TO_SECONDS * 3
+    # recent filled allowed time delay to consider as pending order_filled callback
+    RECENT_TRADES_ALLOWED_TIME = 10
+    # when True, orders creation/health check will be performed on start()
+    SCHEDULE_ORDERS_CREATION_ON_START = True
+
+    def __init__(self, channel, config, trading_mode, exchange_manager):
+        super().__init__(channel, config, trading_mode, exchange_manager)
+        # no state for this evaluator: always neutral
+        self.state = EvaluatorStates.NEUTRAL
+        self.symbol = trading_mode.symbol
+        self.symbol_market = None
+        self.min_max_order_details = {}
+        fees = get_fees(exchange_manager, self.symbol)
+        self.max_fees = max(fees[ExchangeConstantsMarketPropertyColumns.TAKER.value],
+                            fees[ExchangeConstantsMarketPropertyColumns.MAKER.value])
+        self.flat_increment = None
+        self.flat_spread = None
+        self.current_price = None
+
+        self.healthy = False
+
+        # used not to refresh orders when order_fill_callback is processing
+        self.lock = Lock()
```

Comment:
Can this lock be an issue during backtesting?

Reply:
It should not be, it's to prevent concurency accesses when creating orders, that can't happen on backtesting

Reply:
ok :+1: